zwave_js_server/util/node.py

```diff
@@ -0,0 +1,106 @@
+    # we can't use value ID
+    if isinstance(property_or_property_name, str):
+        try:
+            zwave_value = next(
+                config_value
+                for config_value in config_values.values()
+                if config_value.property_name == property_or_property_name
+            )
+        except StopIteration:
+            raise NotFoundError(
+                "Configuration parameter with parameter name "
+                f"{property_or_property_name} could not be found"
+            ) from None
+    else:
+        value_id = get_value_id(
+            node,
+            CommandClass.CONFIGURATION,
+            property_or_property_name,
+            endpoint=0,
+            property_key=property_key,
+        )
+
+        try:
+            zwave_value = config_values[value_id]
+        except KeyError:
+            raise NotFoundError(
+                f"Configuration parameter with value ID {value_id} could not be "
+                "found"
+            ) from None
+
+    # Validate that new value for enumerated configuration parameter is a valid state
+    # key or label
+    if (
+        zwave_value.configuration_value_type == ConfigurationValueType.ENUMERATED
+        and str(new_value)
+        not in [
+            *zwave_value.metadata.states,
+            *zwave_value.metadata.states.values(),
+        ]
+    ):
+        raise InvalidNewValue(
+            "Must provide a value that represents a valid state key or label from "
+            f"{json.dumps(zwave_value.metadata.states)}"
+        )
+
+    # If needed, convert a state label to its key. We know the state exists because
+    # of the validation above.
+    if isinstance(new_value, str):
+        new_value = int(
+            next(
+                key
+                for key in zwave_value.metadata.states
+                if zwave_value.metadata.states[key] == new_value
+            )
+        )
+
+    if zwave_value.configuration_value_type == ConfigurationValueType.UNDEFINED:
+        # We need to use the Configuration CC API to set the value for this type
+        raise NotImplementedError("Configuration values of undefined type can't be set")
+
+    # Validate that new value for range configuration parameter is within bounds
+    max_ = zwave_value.metadata.max
+    min_ = zwave_value.metadata.min
+    if zwave_value.configuration_value_type == ConfigurationValueType.RANGE and (
+        (max_ is not None and new_value > max_)
+        or (min_ is not None and new_value < min_)
+    ):
+        bounds = []
+        if min_ is not None:
+            bounds.append(f"Min: {min_}")
+        if max_ is not None:
+            bounds.append(f"Max: {max_}")
+        raise InvalidNewValue(
+            f"Must provide a value within the target range ({', '.join(bounds)})"
+        )
+
+    # Finally attempt to set the value and return the Value object if successful
+    if not await node.async_set_value(zwave_value, new_value):
+        return None
```

Comment:
I'm thinking if we should raise an error in this case instead?

I also realize now that we don't check this return value in the core integration. We probably should do that.

Here's when it can return False:
> It resolves to false if any of the following conditions are met:
>
> - The setValue API is not implemented in the required Command Class
> - The required Command Class is not implemented in this library yet
> - The API for the required Command Class is not implemented in this library yet

Reply:
Should we raise errors for each condition in `async_set_value` instead? Can we detect each of these scenarios?

Reply:
It's just a general False return value we get from the server. I don't think we can detect details more than that.

https://zwave-js.github.io/node-zwave-js/#/api/node?id=setvalue

Reply:
OK 👍🏾

Reply:
do we want to raise an error if false? or just add handling on the integration side?

Reply:
The return value of the function will be cleaner if we raise an error.

Long term we can change `node.async_set_value` to raise this error instead.